### PR TITLE
feat(analytics): google analytics integrate

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/kittenTricks/blob/master/CONTRIBUTING.md) guide.

### CODE REVIEW

**Comment by artyorsh** [comment]
wow 👀

File: src/app.component.tsx [hunk 1]
```diff
@@ -33,13 +35,38 @@ const assets: Assets = {
 
 export default class App extends React.Component {
 
+  private getActiveRouteName = (navigationState): string | null => {
```
Comment (artyorsh) [inline src/app.component.tsx]: we have this code somewhere in navigation utils

File: src/app.component.tsx [hunk 2]
```diff
@@ -33,13 +35,38 @@ const assets: Assets = {
...
+
+  private onNavigationStateChange = (prevState: NavigationState,
+                                     currentState: NavigationState): void => {
+
+    const prevStateName: string | null = this.getActiveRouteName(prevState);
+    const currentStateName: string | null = this.getActiveRouteName(currentState);
+    if (prevStateName !== currentStateName) {
+      trackScreenTransition(currentStateName)
+        // .then(() => console.log('success'))
+        .catch((error: any) => console.warn('Analytics error: ', error.message));
```
Comment (artyorsh) [inline src/app.component.tsx]: wrap this into a function like `onTransitionTrackError`

**Review by artyorsh — APPROVED** [review]

File: src/app.component.tsx (hunk 1 above)
Comment (32penkin) [reply src/app.component.tsx]: Done.

File: src/app.component.tsx (hunk 2 above)
Comment (32penkin) [reply src/app.component.tsx]: Done.

**Pull request merged**